build/entrypoint.sh

```diff
@@ -2,34 +2,67 @@
+        echo "You do not own the data in $HOME. Please chown it to $(id -u), run the container as the owner of the data or run the container as root."
+        exit 1
+    fi
+fi
 
-    # Initialize pass
+if [[ ! -f $HOME/.gnupg ]]; then
+    echo "No GPG key found in $HOME/.gnupg. Running gpg --generate-key."
     gpg --generate-key --batch /protonmail/gpgparams
     pass init pass-key
-    
-    # Kill the other instance as only one can be running at a time.
-    # This allows users to run entrypoint init inside a running conainter
-    # which is useful in a k8s environment.
-    # || true to make sure this would not fail in case there is no running instance.
-    pkill protonmail-bridge || true
-
-    # Login
-    /protonmail/proton-bridge --cli $@
+fi
+# delete lock files if they exist - this can happen if the container is restarted forcefully
 
-else
+if [[ `find $HOME -name "*.lock" | wc -l` != 0 ]]; then
+    echo "Deleting lock files in $HOME. This can happen if the container is restarted forcefully."
+    find $HOME -name "*.lock" -delete
+fi
 
-    # socat will make the conn appear to come from 127.0.0.1
-    # ProtonMail Bridge currently expects that.
-    # It also allows us to bind to the real ports :)
+# socat will make the conn appear to come from 127.0.0.1
+# ProtonMail Bridge currently expects that.
+# It also allows us to bind to the real ports :)
+if [[ $(id -u) == 0 ]]; then
     socat TCP-LISTEN:25,fork TCP:127.0.0.1:1025 &
     socat TCP-LISTEN:143,fork TCP:127.0.0.1:1143 &
+else
+    socat TCP-LISTEN:2025,fork TCP:127.0.0.1:1025 &
+    socat TCP-LISTEN:2143,fork TCP:127.0.0.1:1143 &
+fi
 
-    # Start protonmail
-    # Fake a terminal, so it does not quit because of EOF...
-    rm -f faketty
-    mkfifo faketty
-    cat faketty | /protonmail/proton-bridge --cli $@
+# Broken until https://github.com/ProtonMail/proton-bridge/issues/512 is resolved.
+# check if the vault-editor can read the config
+/protonmail/vault-editor read 2>&1 1>/dev/null
+# Modify the protonmail config with env variables and expected values. Env variables must be converted from string to boolean.
+/protonmail/vault-editor read | \
+jq '.Settings.AutoUpdate = (env.PROTONMAIL_AutoUpdate | if . == "true" then true else false end)
+| .Settings.TelemetryDisabled = (env.PROTONMAIL_TelemetryDisabled | if . == "true" then true else false end)
+| .Settings.GluonDir |= "\(env.HOME)/.local/share/protonmail/bridge-v3/gluon"
+| .Settings.Autostart = false
+| .Settings.SMTPPort = 1025
+| .Settings.IMAPPort = 1143 ' \
+| /protonmail/vault-editor write
 
-fi
+# Start protonmail
+echo "Starting ProtonMail Bridge. Connect to the CLI with `docker exec -it <container_name>` and type 'help' for instructions."
+/protonmail/proton-bridge --cli $@
+echo "ProtonMail bridge stopped. waiting 30 seconds before exiting in order to preserve the logs."
+sleep 30 # so we have time to read the logs in case of a crash loop
```

Comment:
Wouldn't it be better to delete any remaining lock files at the end of that script too? For some reason the bridge seems to not clean those sometimes for me. But in any case at the last line of the entrypoint any remaining lock file has to have been forgotten and can be deleted safely no?